include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "HttpInvalidRequestLine";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_meek_fronted_requests(settings::MeekFrontedRequestsSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "MeekFrontedRequests";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_multi_ndt(settings::MultiNdtSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "MultiNdt";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_ndt(settings::NdtSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "Ndt";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_tcp_connect(settings::TcpConnectSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "TcpConnect";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_telegram(settings::TelegramSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "Telegram";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_web_connectivity(settings::WebConnectivitySettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "WebConnectivity";
+    /* No nettest specific settings */
+    return start_common(std::move(doc), settings, router);
+}
+
+bool Nettest::start_whatsapp(settings::WhatsappSettings settings, routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    nlohmann::json doc;
+    doc["name"] = "Whatsapp";
+    doc["options"]["all_endpoints"] = (int64_t)settings.all_endpoints;
+    return start_common(std::move(doc), settings, router);
+}
+
+void Nettest::route_next_event(routers::Router *router) noexcept {
+    router = (router) ? router : default_router();
+    UniqueEvent eventptr;
+    nlohmann::json ev;
+    {
+        std::unique_lock<std::mutex> _{mutex_};
+        eventptr.reset(mk_task_wait_for_next_event(task_.get()));
+    }
+    if (eventptr == nullptr) {
+        // TODO(bassosimone): route this error.
+        interrupt();
```

Comment:
The design decision here is that anytime we see something unexpected we log and interrupt the test.